### PR TITLE
Remove thread pool

### PR DESCRIPTION
**Pull request by julianoes**
Before introducing the callback queue, we used a thread pool of 3 threads to call user callbacks.
The thread pool was written to avoid everything from stalling if a user blocked in one of the callbacks.
It generally worked provided the user did not block in more than 3 callbacks at any given time.

However, it had also some disadvantages:

- Due to the thread pool multiple threads could call a function at any given time.
  This means that one callback can be called from at least two threads at the same time which is quite likely if the callback is called at a high rate.
  This can create race conditions on the user side unless a mutex is used to prevent this.
  However, it's not very intuitive that a mutex is needed for that, and mutexes in general should be avoided, if possible, for performance reasons.
  Such race conditions also showed up as CI test failures, see: [MAVSDK#1010](https://github.com/mavlink/MAVSDK/issues/1010), [MAVSDK#1045](https://github.com/mavlink/MAVSDK/issues/1045).
- For developers the old thread pool behaviour was unclear and unpredictable.
  Basically, without code inspection, it was difficult to know if blocking in a callback is safe or not.
  When trying with one blocking callback, it would work, however, this can create a false sense of security as with 3 blocking callbacks it would suddenly lock up.
- Previously, when this happened, it was difficult to debug because it would either lead to a crash on destruction, or lock up at which point the backtrace of all threads need to be looked at using gdb.

### CODE REVIEW

**Comment by julianoes** [comment]
Alright, let's get this in.
